js/index.js

```diff
@@ -1,23 +1,35 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  const priceValue = price.innerHTML;
+  const quantityValue = quantity.value;
+  const subTotalPrice = priceValue * quantityValue;
+  const subTotal = product.querySelector('.subtotal span');
+  const subTotalCalc = (subTotal.innerHTML = subTotalPrice);
+
+  return subTotalPrice;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
-
-  // ITERATION 3
-  //... your code goes here
+  let productValue = document.querySelectorAll('.product');
+  productValue = [...productValue];
+
+  let totalValue = 0;
+  productValue.forEach((element) => {
+    updateSubtotal(element);
+    totalValue += updateSubtotal(element);
```

Comment:
line 21 is unnesessary, since we are already getting the result of this directly on line 22.